### PR TITLE
feat: get form by key - service + endpoint

### LINKED ISSUE #22449: Implement `Get Form for User Task`
### Description
Implement the `GET /v2/user-tasks/{userTaskKey}/form` endpoint for retrieving a form by user task key. This endpoint should return detailed form information including tenant ID, BPMN ID, schema, version, and more.

---

### Endpoint
`GET /v2/user-tasks/{userTaskKey}/form`

#### Request
- **Path Parameter:**
  - `userTaskKey` (Long): The unique key for User Task Key 

#### Response
- **Success (200 OK):**
  - Returns a JSON object representing the form with the following fields:
    - `formKey`: The unique key of the form.
    - `tenantId`: The tenant ID associated with the form.
    - `bpmnId`: The BPMN ID of the form.
    - `schema`: The schema representing the form.
    - `version`: The version of the form.

  **Example Response:**
  ```json
  {
    "formKey": 1,
    "tenantId": "tenant-1",
    "bpmnId": "bpmn-1",
    "schema": {},
    "version": 1
  }

Tasks:
- Implement the Service
- Implement the Rest endpoint
- Implement the Client
- Implement the integration Tests

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Implement the Service and Gateway Rest for Get Form by Key

GET:
`/v2/forms/{formKey}`

Considerations
- The response is set as key, but on the indice tasklist-form it is using "id", because of the compatibility, thats the reason in backend we are querying by "id"
- In order to not breaking the default filter, `key` was added to `tasklist-form` indice, but meanwhile data is not migrated, `id` will be queried from `tasklist-form`, and only Long will be accepted. Which means only **Camunda Forms** will be returned. (And not embedded) as expected by the requirement.

## Related issues

closes #https://github.com/camunda/camunda/issues/22449

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -1564,7 +1564,63 @@ paths:
...
+  /forms:
+    get:
+      tags:
+        - Forms
+      summary: Return form by a Form Key
+      description: |
+        Get the form by the form key.
+
+        :::note
+        This endpoint will return only linked foms. Embedded forms are not supported.
```
Comment (ralfpuchert) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1576]: Typo `forms`. We need also a period after the sentence. For example:

`summary: Return form by a Form Key.`
Could you check other descriptions for the `.` ?

**Review by ralfpuchert — APPROVED** [review]
Only minor comments. Other then that LGTM!

**Review by christinaausley — APPROVED** [review]

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -1564,7 +1564,63 @@ paths:
             application/problem+json:
               schema:
                 $ref: "#/components/schemas/ProblemDetail"
+  /forms:
+    get:
+      tags:
+        - Forms
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1570]: ❌ Please use singular terms in the tags, so `Form` in this case.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/FormController.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    } catch (final Exception e) {
+      REST_LOGGER.warn("An exception occurred in get Form by key.", e);
+      final var problemDetail =
+          RestErrorMapper.createProblemDetail(
+              HttpStatus.INTERNAL_SERVER_ERROR,
+              e.getMessage(),
+              "Failed to execute Get Form by key.");
+      return RestErrorMapper.mapProblemToResponse(problemDetail);
+    }
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/FormController.java:53]: See [comment](https://github.com/camunda/camunda/pull/22480/files#r1768043590) in `Get task by key`.

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/FormControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      {
+        "key": 1,
+        "tenantId": "tenant-1",
+        "bpmnId": "bpmn-1",
+        "schema": "schema",
+        "version": 1
+      }
+      """;
+
+  @Autowired private MockMvc mockMvc;
```
Comment (tmetzke) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/FormControllerTest.java:46]: ❌ The `RestControllerTest` already provides a `WebClient` you should be able to use in the tests.

**Comment by nathansandi** [comment]
I'm going to update the branch this night, after I merge Get Task by Key with the rest error updates.

**Pull request merged**